### PR TITLE
Early development

### PR DESCRIPTION
**Pull request by Baggerone**

### CODE REVIEW

File: cron/alerts/main.go [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+		log.Print("-----")
+		results[zoneName] = subRecs
+	}
+
+}
+
+func getCFRecords(config AlertsConfig) (map[string][]string, error) {
+	api, err := cloudflare.New(config.CFApiKey, config.CFApiEmail)
+	if err != nil {
+		log.Fatal(err)
```
Comment (Schparky) [inline cron/alerts/main.go:198]: Not in scope for this PR, but have you considered sending an email for this and other failures? This is a set-and-forget kind of app so notifying on error could be important.

File: cron/alerts/main.go [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+	CFZoneNames      []string `json:"CFZoneNames"`
+	CFContainsStrings []string   `json:"CFContainsString"`
+	SESAWSRegion     string   `json:"SESAWSRegion"`
+	SESCharSet       string   `json:"SESCharSet"`
+	SESReturnToAddr  string   `json:"SESReturnToAddr"`
+	SESSubjectText   string   `json:"SESSubjectText"`
+	RecipientEmails  []string `json:"RecipientEmails"`
+}
+
+func getCFRecordsWithSubstring(substring, zoneName string, recs []cloudflare.DNSRecord, results map[string][]string) {
```
Comment (Schparky) [inline cron/alerts/main.go:178]: Do you need a test for `getCFRecordsWithSubstring`?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,22 @@
+FROM golang:latest
+
+ENV GO111MODULE on
+
+# Install packages
+RUN curl -sL https://deb.nodesource.com/setup_10.x | bash -
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply Dockerfile:6]: :+1:

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+aws.env
+*.aes
+local.env
+.env
+
+# dev tools metadata
+.idea
+
+# go modules
+go.sum
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply .gitignore:19]: I was confusing it with something else I think.  It looks like it is better to include it.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,22 @@
+FROM golang:latest
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply Dockerfile:1]: :+1:

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+ENV GO111MODULE on
+
+# Install packages
+RUN curl -sL https://deb.nodesource.com/setup_10.x | bash -
+RUN apt-get install -y git nodejs netcat
+#RUN go get -u github.com/golang/dep/cmd/dep
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply Dockerfile:8]: :+1:

File: codeship-steps.yml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+- name: test
+  service: app
+  command: ./codeship/test.sh
+
+- name: deploy_prod
+  service: app
+  tag: master
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply codeship-steps.yml:7]: :+1:

File: cron/alerts/main.go [hunk 8]
```diff
@@ -0,0 +1,330 @@
...
+	gluedSearchStrings := os.Getenv(EnvKeyCFContainsStrings)
+	if gluedSearchStrings == "" {
+		return fmt.Errorf("required value missing for environment variable %s", EnvKeyCFContainsStrings)
+	}
+
+	a.CFContainsStrings = splitStringList(gluedSearchStrings)
+	return nil
+}
+
+func getRequiredString(envKey string, configEntry *string) error {
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply cron/alerts/main.go:105]: Maybe I can get back to this later.  I'm hoping to get it functioning completely today before I leave for CA.

File: cron/alerts/main.go [hunk 9]
```diff
@@ -0,0 +1,330 @@
...
+
+	log.Printf("Starting scan for %v", config.CFZoneNames)
+
+	results := map[string][]string{}
+
+	for _, zoneName := range config.CFZoneNames {
+		zoneID, err := api.ZoneIDByName(zoneName)
+		if err != nil {
+			err = fmt.Errorf("error getting Cloudflare zone %s ... %v ", zoneName, err.Error())
+			return results, err
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply cron/alerts/main.go:209]: Good idea.  I'll try to get back to this.

File: cron/alerts/main.go (hunk 1 above)
Comment (Baggerone) [reply cron/alerts/main.go:198]: Good idea.  I'll try to get back to this.

File: cron/alerts/main.go (hunk 2 above)
Comment (Baggerone) [reply cron/alerts/main.go:178]: Yep. I'll try to get back to this.

**Pull request merged**